picasso/src/main/java/com/squareup/picasso3/Picasso.java

```diff
@@ -751,14 +751,7 @@ public Builder withCacheSize(int maxByteCount) {
       if (maxByteCount < 0) {
         throw new IllegalArgumentException("maxByteCount < 0: " + maxByteCount);
       }
-      if (cache != null) {
-        throw new IllegalStateException("Memory cache already set.");
-      }
-      if (maxByteCount == 0) {
-        cache = Cache.NONE;
-      } else {
-        cache = new LruCache(maxByteCount);
-      }
+      cache = new PlatformLruCache(maxByteCount != 0 ? maxByteCount : 1);
```

Comment:
Let's move this ternary inside `PlatformLruCache` as it's an implementation detail of that type.

Reply:
Will do. Will update the test with 0, then, too.

Reply:
updated and updated tests. I didn't add a test for this in PlatformLruCacheTest, since it's not desired behavior, exactly. Let me know if I should!

Reply:
just curious: what's not desired behavior?  either way, sgtm.

Reply:
I mean that I could have had a test showing that a 0-size cache can really hold a 1sqr alpha_8. But, that's not desired behavior. It'd be a test of a workaround.

Reply:
ah got it.